### DISCUSSION TITLE
PAG动效播放崩溃

### DISCUSSION THREAD

**Discussion by DXBbeginner** [Tencent/libpag#1978] [Q&A]
【版本信息】
PAG SDK 版本 4.3.33

【平台信息】
Android 原生 

【预期的表现】
页面发生一次点击事件创建一个PAGImageView正常播放动效

【实际的情况】
切换页面，父容器清除所有正在播放的PAGImageView，切回页面，再次点击页面，出现崩溃。

12-11 14:30:35.037 15868 17390 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x3 in tid 17390 (tgfx_JNIEnviron), pid 15868 (20.1_0137245054)
12-11 14:30:36.252 18259 18259 F DEBUG   : keyProcess: 0
12-11 14:30:36.252 18259 18259 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
12-11 14:30:36.252 18259 18259 F DEBUG   : Build fingerprint: 'OnePlus/MT2110_CH/OP5154L1:12/RKQ1.211119.001/R.202212211617:user/release-keys'
12-11 14:30:36.252 18259 18259 F DEBUG   : Revision: '0'
12-11 14:30:36.252 18259 18259 F DEBUG   : ABI: 'arm'
12-11 14:30:36.252 18259 18259 F DEBUG   : Timestamp: 2023-12-11 14:30:35.309400073+0800
12-11 14:30:36.252 18259 18259 F DEBUG   : Process uptime: 0s
12-11 14:30:36.252 18259 18259 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x3
12-11 14:30:36.252 18259 18259 F DEBUG   : Cause: null pointer dereference
12-11 14:30:36.252 18259 18259 F DEBUG   :     r0  ff000100  r1  ffffffff  r2  a4203cc2  r3  0027c2ff
12-11 14:30:36.252 18259 18259 F DEBUG   :     r4  00000001  r5  ffffffff  r6  a4203cc9  r7  69ffbab8
12-11 14:30:36.252 18259 18259 F DEBUG   :     r8  00000000  r9  0000001f  r10 a420ab4a  r11 a420ab40
12-11 14:30:36.252 18259 18259 F DEBUG   :     ip  0027c2f3  sp  69ffba70  lr  ffffffff  pc  58351058
12-11 14:30:36.252 18259 18259 F DEBUG   : backtrace:
12-11 14:30:36.252 18259 18259 F DEBUG   :       #00 pc 000c0058  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #01 pc 00079239  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #02 pc 00067541  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (pag::PAGDecoder::readFrameInternal(int, std::__ndk1::shared_ptr<pag::BitmapBuffer>)+372) (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #03 pc 000677cb  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (pag::PAGDecoder::readFrame(int, AHardwareBuffer*)+78) (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #04 pc 000b76fd  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (Java_org_libpag_PAGDecoder_readFrame+56) (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #05 pc 0017e08d  /apex/com.android.art/lib/libart.so (art_quick_generic_jni_trampoline+44) (BuildId: fe65c9191eeeb05ab1a462c3b4b557ab)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #06 pc 000f1b8c  /apex/com.android.art/lib/libart.so (nterp_helper+2908) (BuildId: fe65c9191eeeb05ab1a462c3b4b557ab)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #07 pc 0046992e  [anon:dalvik-classes26.dex extracted in memory from /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/com.baidu.searchbox-sFelToJWL9QLBFZoKZgpgw==/base.apk!classes26.dex] (org.libpag.c$a.a+14)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #08 pc 02466cad  /memfd:jit-cache (deleted) (org.libpag.PAGImageView.a+1524)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #09 pc 024884eb  /memfd:jit-cache (deleted) (org.libpag.PAGImageView.flush+490)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #10 pc 02498d31  /memfd:jit-cache (deleted) (org.libpag.PAGImageView.onAnimationUpdate+248)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #11 pc 000f253c  /apex/com.android.art/lib/libart.so (nterp_helper+5388) (BuildId: fe65c9191eeeb05ab1a462c3b4b557ab)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #12 pc 00465b0c  [anon:dalvik-classes26.dex extracted in memory from /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/com.baidu.searchbox-sFelToJWL9QLBFZoKZgpgw==/base.apk!classes26.dex] (org.libpag.PAGAnimator.onAnimationUpdate+20)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #13 pc 001795d5  /apex/com.android.art/lib/libart.so (art_quick_invoke_stub_internal+68) (BuildId: fe65c9191eeeb05ab1a462c3b4b557ab)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #14 pc 001916e7  /apex/com.android.art/lib/libart.so (void art::quick_invoke_reg_setup<false>(art::ArtMethod*, unsigned int*, unsigned int, art::Thread*, art::JValue*, char const*) (.llvm.4852100798313944515)+234) (BuildId: fe65c9191eeeb05ab1a462c3b4b557ab)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #15 pc 001ab177  /apex/com.android.art/lib/libart.so (art::JValue art::InvokeVirtualOrInterfaceWithVarArgs<art::ArtMethod*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, art::ArtMethod*, std::__va_list)+702) (BuildId: fe65c9191eeeb05ab1a462c3b4b557ab)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #16 pc 001aae99  /apex/com.android.art/lib/libart.so (art::JValue art::InvokeVirtualOrInterfaceWithVarArgs<_jmethodID*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+24) (BuildId: fe65c9191eeeb05ab1a462c3b4b557ab)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #17 pc 003f27f5  /apex/com.android.art/lib/libart.so (art::JNI<true>::CallVoidMethodV(_JNIEnv*, _jobject*, _jmethodID*, std::__va_list)+196) (BuildId: fe65c9191eeeb05ab1a462c3b4b557ab)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #18 pc 00370a57  /apex/com.android.art/lib/libart.so (art::(anonymous namespace)::CheckJNI::CallMethodV(char const*, _JNIEnv*, _jobject*, _jclass*, _jmethodID*, std::__va_list, art::Primitive::Type, art::InvokeType)+1058) (BuildId: fe65c9191eeeb05ab1a462c3b4b557ab)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #19 pc 003727f5  /apex/com.android.art/lib/libart.so (art::(anonymous namespace)::CheckJNI::CallVoidMethodV(_JNIEnv*, _jobject*, _jmethodID*, std::__va_list) (.llvm.8452170382640045565)+44) (BuildId: fe65c9191eeeb05ab1a462c3b4b557ab)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #20 pc 000b32b1  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #21 pc 000b5e81  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #22 pc 00065d11  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #23 pc 00066531  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #24 pc 000e861b  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #25 pc 000e8821  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #26 pc 000e8d91  /data/app/~~QpO8MZwla0_kE3ANGyQQ6A==/xxxxx-sFelToJWL9QLBFZoKZgpgw==/lib/arm/libpag.so (BuildId: b56c9d2ce77abb66a258bed0280127baf82dc890)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #27 pc 0009f9d7  /apex/com.android.runtime/lib/bionic/libc.so (__pthread_start(void*)+40) (BuildId: ef8d01bc76dae667c4c8eb4c932f4e45)
12-11 14:30:36.252 18259 18259 F DEBUG   :       #28 pc 00058933  /apex/com.android.runtime/lib/bionic/libc.so (__start_thread+30) (BuildId: ef8d01bc76dae667c4c8eb4c932f4e45)

**Comment by CodeJhF** [comment]
抱歉这条当时没有被跟进到，是我们漏了。

因长期无回复，且 libpag 4.3 及更早版本已停止维护（当前仅维护 4.4 / 4.5），先关闭本条。
上报版本：4.3.33 / 1.211119.001。

如果问题在最新版本（4.4 / 4.5）仍能复现，欢迎在本帖回复或新开 issue，并附上 libpag 版本号、平台与系统版本、可复现的 PAG 文件或最小 demo，我们会优先跟进。（stale-close）
